dsp-user/repo/src/test/scala/dsp/user/repo/impl/UserRepoImplSpec.scala

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright © 2021 - 2022 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package dsp.user.repo.impl
+
+import zio.ZLayer
+import zio._
+import zio.test._
+import dsp.user.domain.User
+import dsp.user.repo.impl.UserRepoLive
+import dsp.user.repo.impl.UserRepoMock
+import zio.test.ZIOSpecDefault
+import dsp.user.api.UserRepo
+import dsp.valueobjects.User._
+import dsp.errors.BadRequestException
+import zio.prelude.Validation
+import zio.prelude.ZValidation
+
+/**
+ * This spec is used to test all [[dsp.user.repo.UserRepo]] implementations.
+ */
+object UserRepoImplSpec extends ZIOSpecDefault {
+
+  def spec = (userRepoMockTests + userRepoLiveTests)
+
+  private val testUser1 = (for {
```

Comment:
This would be a good occasion to create some shared test-data for the user entity.